src/Server/FableJson.fs

```diff
@@ -14,3 +15,15 @@ let toJson value =
 
 let ofJson<'a> (json:string) : 'a =
     JsonConvert.DeserializeObject<'a>(json, [|jsonConverter|])
+
+let serialize x =
+    setStatusCode 200
+      >=> setHttpHeader "Content-Type" "application/json"
+      >=> setBodyAsString (toJson x)
```

Comment:
I actually like the Giraffe idea to make it a habit of using negotiation for the responses instead of always using `application/json`. But we can't use Giraffe's default `negotiate` because it doesn't use Fable JSON serialization so we have a custom one:

```fsharp
  let fableJson (dataObj : obj) : HttpHandler =
    setHttpHeader "Content-Type" "application/json"
    >=> setBodyAsString (FableJson.toJson dataObj)

  let negotiate (responseObj : obj) : HttpHandler =
    negotiateWith
        // Default negotiation rules
        (dict [
            "*/*"             , fableJson
            "application/json", fableJson
            "application/xml" , xml
            "text/xml"        , xml
            "text/plain"      , string >> text
        ])
        // Default unacceptable HttpHandler
        (fun (next : HttpFunc) (ctx : HttpContext) ->
            (setStatusCode 406
            >=> ((ctx.Request.Headers.["Accept"]).ToString()
                |> sprintf "%s is unacceptable by the server."
                |> text)) next ctx)
        // response object
        responseObj
```

Then, for example:

```fsharp
return! Successful.ok (Http.negotiate wishList) next ctx
```

Reply:
we should patch that into giraffe - or create a giraffe.fable or something

Reply:
Already too many packages ;) I guess adding a `negotiateWithJson` or similar to Giraffe so you can pass a custom Json serializer would work :+1:

Reply:
https://github.com/giraffe-fsharp/Giraffe/pull/179 this would allow us to patch the negotiation rules easily.